### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by sikender**

### CODE REVIEW

File: docs/examples/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const Apigee = require("../../dist/index");
+const apigee = new Apigee("username", "password");
+// const apigee = new Apigee(
```
Comment (ravindrasingh22) [inline docs/examples/index.js:3]: remove the commented code if not needed.

File: package.json [hunk 2]
```diff
@@ -1,17 +1,17 @@
 {
-  "name": "apigee-node-sdk",
-  "description": "Apigee Management API Node.js SDK",
+  "name": "apigee-management-oauth",
```
Comment (ravindrasingh22) [inline package.json:2]: To publish this on npm name should be @srijanone/apigee-management-oauth

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import AuthService from "./auth.service";
+
+export default class Apigee {
+  private apigee: AuthService;
+
+  constructor(
+    username: string,
+    password: string,
+    loginUrl = "https://login.apigee.com/oauth",
+    oauthUsername = "edgecli",
```
Comment (ravindrasingh22) [inline src/index.ts:11]: username and password are common for all? shouldn't it be parameterized? and pick from env.

**Review by ravindrasingh22 — COMMENTED** [review]
Please have a look into the high level comments as of now.

File: src/index.ts (hunk 3 above)
Comment (sikender) [reply src/index.ts:11]: This is TypeScript optional parameter syntax. If the credentials aren't passed to the constructor then these will be used.

File: docs/examples/index.js (hunk 1 above)
Comment (sikender) [reply docs/examples/index.js:3]: This is an example of how to instantiate the module with additional info.

File: package.json (hunk 2 above)
Comment (sikender) [reply package.json:2]: Will update

**Review by ravindrasingh22 — APPROVED** [review]

**Pull request merged**